src/core/statistics_chain.cpp

```diff
@@ -64,26 +63,27 @@ void calc_re(PartCfg &partCfg, double **_re) {
     dist2 += tmp;
     dist4 += tmp * tmp;
   }
-  tmp = (double)chain_n_chains;
+  tmp = static_cast<double>(chain_n_chains);
   re[0] = dist / tmp;
   re[2] = dist2 / tmp;
   re[1] = sqrt(re[2] - re[0] * re[0]);
   re[3] = sqrt(dist4 / tmp - re[2] * re[2]);
+  return re;
 }
 
-void calc_rg(PartCfg &partCfg, double **_rg) {
-  int i, j, p;
+std::vector<double> calc_rg(PartCfg &partCfg) {
```

Comment:
Shouldn't be on the heap, the size is known.

Reply:
yep